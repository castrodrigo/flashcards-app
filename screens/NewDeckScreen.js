import React from "react";
import { Keyboard } from "react-native";
import styled from "styled-components/native";
import Header from "../components/Header";
import Container from "../components/Layout/Container";
import Title from "../components/Layout/Title";
import Description from "../components/Layout/Description";
import Button, { ButtonLabel } from "../components/Layout/Button";
import Form from "../components/Form/Form";
import Input from "../components/Form/Input";
import ErrorMessage from "../components/Form/ErrorMessage";

const SaveButton = styled(Button)`
  background: #a560e7;
  border-bottom-color: #b881ec;
`;

export default class NewDeckScreen extends React.Component {
  static navigationOptions = {
    header: <Header>New Deck</Header>
  };

  state = {
    name: "",
    description: "",
    error: false
  };

  handleSubmit = () => {
    const { name, description } = this.state;
    if (name !== "") {
      return this.props.onSubmit({ name, description });
    }
    return this.setState({ error: true });
  };

  handleChange = (data, field) =>
    this.setState({ [field]: data, error: false });

  render() {
    return (
      <Container>
        <Title>Creating a new Deck</Title>
        <Description>
          You can create a Deck to group a collection of topics that you would
          like to revisit. It is possible to trigger a Quiz through all the
          inserted questions.
        </Description>
        <Form>
          <Input
            placeholder="Name"
            maxLength={30}
            onBlur={Keyboard.dismiss}
            onChangeText={e => this.handleChange(e, "name")}
          />
          <Input
            placeholder="Description"
            maxLength={60}
            onBlur={Keyboard.dismiss}
            onChangeText={e => this.handleChange(e, "description")}
          />
          {this.state.error && (
            <ErrorMessage>
              Do not forget to define the name of the Deck
            </ErrorMessage>
          )}
          <SaveButton onPress={this.handleSubmit}>
            <ButtonLabel>Create New Deck</ButtonLabel>
          </SaveButton>
        </Form>
      </Container>
    );
  }
}
