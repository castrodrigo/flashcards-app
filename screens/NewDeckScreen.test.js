import React from "react";
import { shallow } from "enzyme";
import NewDeckScreen from "./NewDeckScreen";

const props = {
  onSubmit: jest.fn()
};

describe("Render NewDeckScreen", () => {
  it("should render successfully", () => {
    const wrapper = shallow(<NewDeckScreen {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
