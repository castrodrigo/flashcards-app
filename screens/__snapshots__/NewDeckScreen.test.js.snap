// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render NewDeckScreen should render successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <NewDeckScreen
    onSubmit={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef>
          Creating a new Deck
        </ForwardRef>,
        <ForwardRef>
          You can create a Deck to group a collection of topics that you would like to revisit. It is possible to trigger a Quiz through all the inserted questions.
        </ForwardRef>,
        <ForwardRef>
          <ForwardRef
            maxLength={30}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Name"
            value=""
          />
          <ForwardRef
            maxLength={60}
            onBlur={[Function]}
            onChangeText={[Function]}
            placeholder="Description"
            value=""
          />
          <ForwardRef
            onPress={[Function]}
          >
            <ForwardRef>
              Create New Deck
            </ForwardRef>
          </ForwardRef>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "Creating a new Deck",
        },
        "ref": null,
        "rendered": "Creating a new Deck",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "displayName": "Styled(Component)",
          "inlineStyle": InlineStyle {
            "rules": Array [
              "
  font-size: 24px;
  font-weight: bold;
  color: #1baff6;
",
            ],
          },
          "propTypes": Object {
            "accessible": [Function],
            "adjustsFontSizeToFit": [Function],
            "allowFontScaling": [Function],
            "disabled": [Function],
            "ellipsizeMode": [Function],
            "minimumFontScale": [Function],
            "nativeID": [Function],
            "numberOfLines": [Function],
            "onLayout": [Function],
            "onLongPress": [Function],
            "onPress": [Function],
            "pressRetentionOffset": [Function],
            "selectable": [Function],
            "selectionColor": [Function],
            "style": [Function],
            "suppressHighlighting": [Function],
            "testID": [Function],
            "textBreakStrategy": [Function],
          },
          "render": [Function],
          "styledComponentId": "StyledNativeComponent",
          "target": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "You can create a Deck to group a collection of topics that you would like to revisit. It is possible to trigger a Quiz through all the inserted questions.",
        },
        "ref": null,
        "rendered": "You can create a Deck to group a collection of topics that you would like to revisit. It is possible to trigger a Quiz through all the inserted questions.",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "displayName": "Styled(Component)",
          "inlineStyle": InlineStyle {
            "rules": Array [
              "
  font-size: 14px;
  color: #3c3c3c;
  margin: 10px 0;
  padding: 10px 0;
",
            ],
          },
          "propTypes": Object {
            "accessible": [Function],
            "adjustsFontSizeToFit": [Function],
            "allowFontScaling": [Function],
            "disabled": [Function],
            "ellipsizeMode": [Function],
            "minimumFontScale": [Function],
            "nativeID": [Function],
            "numberOfLines": [Function],
            "onLayout": [Function],
            "onLongPress": [Function],
            "onPress": [Function],
            "pressRetentionOffset": [Function],
            "selectable": [Function],
            "selectionColor": [Function],
            "style": [Function],
            "suppressHighlighting": [Function],
            "testID": [Function],
            "textBreakStrategy": [Function],
          },
          "render": [Function],
          "styledComponentId": "StyledNativeComponent",
          "target": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <ForwardRef
              maxLength={30}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Name"
              value=""
            />,
            <ForwardRef
              maxLength={60}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Description"
              value=""
            />,
            false,
            <ForwardRef
              onPress={[Function]}
            >
              <ForwardRef>
                Create New Deck
              </ForwardRef>
            </ForwardRef>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "maxLength": 30,
              "onBlur": [Function],
              "onChangeText": [Function],
              "placeholder": "Name",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "State": Object {
                "blurTextInput": [Function],
                "currentlyFocusedField": [Function],
                "focusTextInput": [Function],
              },
              "attrs": Array [],
              "displayName": "Styled(TextInput)",
              "getDefaultProps": [Function],
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  border-color: #cccccc;
  border-bottom-width: 1;
  height: 50;
  font-size: 20;
  padding-left: 2px;
  padding-right: 2px;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "maxLength": 60,
              "onBlur": [Function],
              "onChangeText": [Function],
              "placeholder": "Description",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "State": Object {
                "blurTextInput": [Function],
                "currentlyFocusedField": [Function],
                "focusTextInput": [Function],
              },
              "attrs": Array [],
              "displayName": "Styled(TextInput)",
              "getDefaultProps": [Function],
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  border-color: #cccccc;
  border-bottom-width: 1;
  height: 50;
  font-size: 20;
  padding-left: 2px;
  padding-right: 2px;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
          false,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ForwardRef>
                Create New Deck
              </ForwardRef>,
              "onPress": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": "Create New Deck",
              },
              "ref": null,
              "rendered": "Create New Deck",
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "displayName": "Styled(Component)",
                "inlineStyle": InlineStyle {
                  "rules": Array [
                    "
  font-size: 16px;
  font-weight: bold;
  text-align: center;
  text-transform: uppercase;
  color: #fff;
  padding: 0 10px;
  align-self: center;
",
                  ],
                },
                "propTypes": Object {
                  "accessible": [Function],
                  "adjustsFontSizeToFit": [Function],
                  "allowFontScaling": [Function],
                  "disabled": [Function],
                  "ellipsizeMode": [Function],
                  "minimumFontScale": [Function],
                  "nativeID": [Function],
                  "numberOfLines": [Function],
                  "onLayout": [Function],
                  "onLongPress": [Function],
                  "onPress": [Function],
                  "pressRetentionOffset": [Function],
                  "selectable": [Function],
                  "selectionColor": [Function],
                  "style": [Function],
                  "suppressHighlighting": [Function],
                  "testID": [Function],
                  "textBreakStrategy": [Function],
                },
                "render": [Function],
                "styledComponentId": "StyledNativeComponent",
                "target": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "displayName": "Styled(Styled(TouchableOpacity))",
              "getDefaultProps": [Function],
              "inlineStyle": InlineStyle {
                "rules": Array [
                  "
  padding: 10px;
  margin-top: 32px;
  border-radius: 12px;
  border-bottom-width: 3px;
  flex: 1;
  flex-direction: row;
  justify-content: center;
",
                  "
  background: #a560e7;
  border-bottom-color: #b881ec;
",
                ],
              },
              "render": [Function],
              "styledComponentId": "StyledNativeComponent",
              "target": [Function],
              "withComponent": [Function],
              Symbol(Symbol.hasInstance): [Function],
            },
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "displayName": "Styled(Component)",
          "inlineStyle": InlineStyle {
            "rules": Array [
              "
  margin-top: 10px;
  padding: 20px;
  border-width: 1px;
  border-color: #ccc;
  border-radius: 12px;
",
            ],
          },
          "render": [Function],
          "styledComponentId": "StyledNativeComponent",
          "target": [Function],
          "withComponent": [Function],
          Symbol(Symbol.hasInstance): [Function],
        },
      },
    ],
    "type": [MockFunction],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef>
            Creating a new Deck
          </ForwardRef>,
          <ForwardRef>
            You can create a Deck to group a collection of topics that you would like to revisit. It is possible to trigger a Quiz through all the inserted questions.
          </ForwardRef>,
          <ForwardRef>
            <ForwardRef
              maxLength={30}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Name"
              value=""
            />
            <ForwardRef
              maxLength={60}
              onBlur={[Function]}
              onChangeText={[Function]}
              placeholder="Description"
              value=""
            />
            <ForwardRef
              onPress={[Function]}
            >
              <ForwardRef>
                Create New Deck
              </ForwardRef>
            </ForwardRef>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "Creating a new Deck",
          },
          "ref": null,
          "rendered": "Creating a new Deck",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(Component)",
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  font-size: 24px;
  font-weight: bold;
  color: #1baff6;
",
              ],
            },
            "propTypes": Object {
              "accessible": [Function],
              "adjustsFontSizeToFit": [Function],
              "allowFontScaling": [Function],
              "disabled": [Function],
              "ellipsizeMode": [Function],
              "minimumFontScale": [Function],
              "nativeID": [Function],
              "numberOfLines": [Function],
              "onLayout": [Function],
              "onLongPress": [Function],
              "onPress": [Function],
              "pressRetentionOffset": [Function],
              "selectable": [Function],
              "selectionColor": [Function],
              "style": [Function],
              "suppressHighlighting": [Function],
              "testID": [Function],
              "textBreakStrategy": [Function],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "You can create a Deck to group a collection of topics that you would like to revisit. It is possible to trigger a Quiz through all the inserted questions.",
          },
          "ref": null,
          "rendered": "You can create a Deck to group a collection of topics that you would like to revisit. It is possible to trigger a Quiz through all the inserted questions.",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(Component)",
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  font-size: 14px;
  color: #3c3c3c;
  margin: 10px 0;
  padding: 10px 0;
",
              ],
            },
            "propTypes": Object {
              "accessible": [Function],
              "adjustsFontSizeToFit": [Function],
              "allowFontScaling": [Function],
              "disabled": [Function],
              "ellipsizeMode": [Function],
              "minimumFontScale": [Function],
              "nativeID": [Function],
              "numberOfLines": [Function],
              "onLayout": [Function],
              "onLongPress": [Function],
              "onPress": [Function],
              "pressRetentionOffset": [Function],
              "selectable": [Function],
              "selectionColor": [Function],
              "style": [Function],
              "suppressHighlighting": [Function],
              "testID": [Function],
              "textBreakStrategy": [Function],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <ForwardRef
                maxLength={30}
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="Name"
                value=""
              />,
              <ForwardRef
                maxLength={60}
                onBlur={[Function]}
                onChangeText={[Function]}
                placeholder="Description"
                value=""
              />,
              false,
              <ForwardRef
                onPress={[Function]}
              >
                <ForwardRef>
                  Create New Deck
                </ForwardRef>
              </ForwardRef>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "maxLength": 30,
                "onBlur": [Function],
                "onChangeText": [Function],
                "placeholder": "Name",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "State": Object {
                  "blurTextInput": [Function],
                  "currentlyFocusedField": [Function],
                  "focusTextInput": [Function],
                },
                "attrs": Array [],
                "displayName": "Styled(TextInput)",
                "getDefaultProps": [Function],
                "inlineStyle": InlineStyle {
                  "rules": Array [
                    "
  border-color: #cccccc;
  border-bottom-width: 1;
  height: 50;
  font-size: 20;
  padding-left: 2px;
  padding-right: 2px;
",
                  ],
                },
                "render": [Function],
                "styledComponentId": "StyledNativeComponent",
                "target": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "maxLength": 60,
                "onBlur": [Function],
                "onChangeText": [Function],
                "placeholder": "Description",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "State": Object {
                  "blurTextInput": [Function],
                  "currentlyFocusedField": [Function],
                  "focusTextInput": [Function],
                },
                "attrs": Array [],
                "displayName": "Styled(TextInput)",
                "getDefaultProps": [Function],
                "inlineStyle": InlineStyle {
                  "rules": Array [
                    "
  border-color: #cccccc;
  border-bottom-width: 1;
  height: 50;
  font-size: 20;
  padding-left: 2px;
  padding-right: 2px;
",
                  ],
                },
                "render": [Function],
                "styledComponentId": "StyledNativeComponent",
                "target": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
            false,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <ForwardRef>
                  Create New Deck
                </ForwardRef>,
                "onPress": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": "Create New Deck",
                },
                "ref": null,
                "rendered": "Create New Deck",
                "type": Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "attrs": Array [],
                  "displayName": "Styled(Component)",
                  "inlineStyle": InlineStyle {
                    "rules": Array [
                      "
  font-size: 16px;
  font-weight: bold;
  text-align: center;
  text-transform: uppercase;
  color: #fff;
  padding: 0 10px;
  align-self: center;
",
                    ],
                  },
                  "propTypes": Object {
                    "accessible": [Function],
                    "adjustsFontSizeToFit": [Function],
                    "allowFontScaling": [Function],
                    "disabled": [Function],
                    "ellipsizeMode": [Function],
                    "minimumFontScale": [Function],
                    "nativeID": [Function],
                    "numberOfLines": [Function],
                    "onLayout": [Function],
                    "onLongPress": [Function],
                    "onPress": [Function],
                    "pressRetentionOffset": [Function],
                    "selectable": [Function],
                    "selectionColor": [Function],
                    "style": [Function],
                    "suppressHighlighting": [Function],
                    "testID": [Function],
                    "textBreakStrategy": [Function],
                  },
                  "render": [Function],
                  "styledComponentId": "StyledNativeComponent",
                  "target": [Function],
                  "withComponent": [Function],
                  Symbol(Symbol.hasInstance): [Function],
                },
              },
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "displayName": "Styled(Styled(TouchableOpacity))",
                "getDefaultProps": [Function],
                "inlineStyle": InlineStyle {
                  "rules": Array [
                    "
  padding: 10px;
  margin-top: 32px;
  border-radius: 12px;
  border-bottom-width: 3px;
  flex: 1;
  flex-direction: row;
  justify-content: center;
",
                    "
  background: #a560e7;
  border-bottom-color: #b881ec;
",
                  ],
                },
                "render": [Function],
                "styledComponentId": "StyledNativeComponent",
                "target": [Function],
                "withComponent": [Function],
                Symbol(Symbol.hasInstance): [Function],
              },
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "displayName": "Styled(Component)",
            "inlineStyle": InlineStyle {
              "rules": Array [
                "
  margin-top: 10px;
  padding: 20px;
  border-width: 1px;
  border-color: #ccc;
  border-radius: 12px;
",
              ],
            },
            "render": [Function],
            "styledComponentId": "StyledNativeComponent",
            "target": [Function],
            "withComponent": [Function],
            Symbol(Symbol.hasInstance): [Function],
          },
        },
      ],
      "type": [MockFunction],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
